### PR TITLE
Extra summaries

### PR DESCRIPTION
**Pull request by florpi**
Interfaced ltu-cmass diagnosis with ili-summarizer. I left summaries that are fast to compute as default summaries, but there are a couple of odd things:

- My redshift space haloes look odd, like RSD is not doing anything (the quadrupole is basically zero) but I’m not sure if this is due to charm or something else. The RSD implementation should be the same as before: https://github.com/maho3/ltu-cmass/blob/extra_summaries/cmass/diagnostics/tools.py#L6
- Bk gives the strangest errors, but given that Pk looks odd it might be due to the data vector being weird.
- The four summaries computed now take about 3 minutes in a cpu to compute for both configuration and redshift space. It should be faster on a gpu, particularly WST. Density Split is slow, since it computes lots of cross correlations, therefore I left it out for now.
- The binning of each summary statistic is now kind of arbitrary and people should play around with them

### CODE REVIEW

**Comment by DeaglanBartlett** [comment]
Just noting here that the `summarizer` module is needed for this but is not mentioned in the installation notes, so this should be added.

File: cmass/diagnostics/summ.py [hunk 1]
```diff
@@ -72,32 +165,28 @@ def halo_summ(source_path, L, N, h, z, threads=16, from_scratch=True):
...
-                group.create_dataset('zPk_k', data=kz)
-                group.create_dataset('zPk', data=Pkz)
                 group.create_dataset('mass_bins', data=be)
                 group.create_dataset('mass_hist', data=hist)
     return True
 
 
 def gal_summ(source_path, hod_seed, L, N, h, z, threads=16,
-             from_scratch=True):
+             from_scratch=True,  summaries=['Pk','TwoPCF','KNN','WST']):
```
[the remark this answers is not in the document]
Comment (DeaglanBartlett) [reply cmass/diagnostics/summ.py:194]: This has been fixed by adding in a mask such that only halos with all three coordinates in range [0,boxsize) are considered.

[the remark this answers is not in the document]
Comment (florpi) [reply cmass/diagnostics/summ.py:194]: should we wrap them inside the box with %boxsize? why are there haloes outside of the box?

[the remark this answers is not in the document]
Comment (DeaglanBartlett) [reply cmass/diagnostics/summ.py:194]: I just implemented a cut. There weren't really any outside the box when I checked, but there was at least one point with coordinate exactly equal to boxsize, but the kNN code requires all coordinates < this, not <=

File: cmass/diagnostics/summ.py [hunk 2]
```diff
@@ -72,32 +165,33 @@ def halo_summ(source_path, L, N, h, z, threads=16, from_scratch=True):
...
+                hvel = hvel[m]
+                hmass = hmass[m]
+                # Get summaries in comoving space
+                group = o.create_group(a)
+                box_catalogue = get_box_catalogue(pos=hpos, z=z, L=L, N=N)
+                for summ in summaries:
+                    store_summary(box_catalogue, group, summ, L, N, threads, use_rsd=False,)
+
+                # Get summaries in redshift space
+                zbox_catalogue = get_box_catalogue_rsd(pos=hpos, vel=hvel, h=h, z=z, axis=2, L=L, N=N)
```
Comment (DeaglanBartlett) [inline cmass/diagnostics/summ.py:180]: This gives an error for me
```
 File "/home/x-dbartlett/ltu-cmass/cmass/diagnostics/summ.py", line 60, in get_box_catalogue_rsd
    pos = get_redshift_space_pos(pos=pos, vel=vel, z=z, h=h, axis=axis, L=L,)
  File "/home/x-dbartlett/ltu-cmass/cmass/diagnostics/tools.py", line 8, in get_redshift_space_pos
    RSL.pos_redshift_space(pos, vel, L, h*100, z, axis)
  File "library/redshift_space_library/redshift_space_library.pyx", line 29, in redshift_space_library.redshift_space_library.pos_redshift_space
ValueError: Buffer dtype mismatch, expected 'float' but got 'double'
```
Please ensure correct type is passed to the relevant function

**Review by DeaglanBartlett — CHANGES_REQUESTED** [review]

File: cmass/diagnostics/summ.py (hunk 2 above)
Comment (DeaglanBartlett) [reply cmass/diagnostics/summ.py:180]: This has now been fixed by ensuring `pos` and `vel` are of dtype `np.float32`.

**Comment by DeaglanBartlett** [comment]
I have now run the code on the pinocchio simulations with lhid=3 for the quijotelike suite, having run HOD before this:
```bash
python -m cmass.bias.apply_hod nbody=pinocchio_quijote sim=pinocchio
python -m cmass.diagnostics.summ nbody=pinocchio_quijote sim=pinocchio
```

Plotting the results gives the following for the halos:

halos_pk
halos_twopcf
halos_knn
halos_hist

and for galaxies:

galaxies_pk
galaxies_twopcf
galaxies_knn

File: cmass/diagnostics/summ.py [hunk 3]
```diff
@@ -11,9 +11,12 @@
 from omegaconf import DictConfig, OmegaConf
 import h5py
 
+from summarizer.data import BoxCatalogue
+import summarizer
```
[the remark this answers is not in the document]
Comment (DeaglanBartlett) [reply cmass/diagnostics/summ.py:15]: I have included this discussion into the installation instructions for summarizer

**Review by DeaglanBartlett — APPROVED** [review]

**Pull request merged**